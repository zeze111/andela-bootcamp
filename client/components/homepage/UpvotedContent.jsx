import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import ReactTooltip from 'react-tooltip';

/** Stateless component to render details of a recipe card
 *
 * @export {function} UpvotedContent
 *
 * @param {object} recipe
 *
 * @returns {null} null
 */
const UpvotedContent = ({
  recipe: {
    recipeId, upvotes, Recipe
  }
}) => {
  const vote = (
    <span className="home-text">
      <i className="material-icons home-icon">thumb_up</i> {upvotes}
    </span>
  );
  const show = `${Recipe.name.substring(0, 10)}...`;
  return (
    <li >
      <div className="col s2 offset-1">
        <div className="card hoverable grey lighten-4 home-cards">
          <div className="card-image">
            <img
              src={Recipe.image || '/images/noimg.png'}
              alt="recipe"
              className="responsive-img image-style"
            />
            <div className="card-action">
              <Link
                to={`/recipe/${recipeId}`}
                href={`/recipe/${recipeId}`}
                className="home-text-style"
                data-tip={Recipe.name}
              > {show}
                <ReactTooltip />
              </Link> {vote}
            </div>
          </div>
        </div>
      </div>
    </li>

  );
};

UpvotedContent.propTypes = {
  recipe: PropTypes.objectOf(PropTypes.any).isRequired
};

export default UpvotedContent;
