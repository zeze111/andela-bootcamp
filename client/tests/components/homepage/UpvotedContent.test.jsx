import React from 'react';

import UpvotedContent from '../../../components/homepage/UpvotedContent';
import store from '../../../store';

const props = {
  recipe: {
    recipeId: 1,
    upvotes: 20,
    Recipe: {
      name: 'some name'
    }
  }
};

describe('Test for Upvoted Content', () => {
  it('should render component', () => {
    const wrapper = shallow(<UpvotedContent {...props} store={store}/> );
    expect(wrapper).toMatchSnapshot()
  });
});
