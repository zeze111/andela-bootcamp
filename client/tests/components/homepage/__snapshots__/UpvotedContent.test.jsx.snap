// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Upvoted Content should render component 1`] = `
<li>
  <div
    className="col s2 offset-1"
  >
    <div
      className="card hoverable grey lighten-4 home-cards"
    >
      <div
        className="card-image"
      >
        <img
          alt="recipe"
          className="responsive-img image-style"
          src="/images/noimg.png"
        />
        <div
          className="card-action"
        >
          <Link
            className="home-text-style"
            data-tip="some name"
            href="/recipe/1"
            replace={false}
            to="/recipe/1"
          >
             
            some nam...
            <ReactTooltip
              insecure={true}
              resizeHide={true}
              wrapper="div"
            />
          </Link>
           
          <span>
            <i
              className="material-icons home-icon"
            >
              thumb_up
            </i>
             
            20
          </span>
          <span
            className="pop-text"
          >
            <i
              className="material-icons home-icon2"
            >
              visibility
            </i>
          </span>
        </div>
      </div>
    </div>
  </div>
</li>
`;
